java-security/src/test/java/com/sap/cloud/security/token/validation/validators/JwtIssuerValidatorTest.java

```diff
@@ -107,11 +107,38 @@ void validationFails_iasIssuerUrl(String iasIssuer) {
 		assertThat(validationResult.getErrorDescription(), startsWith("Issuer is not trusted because issuer "));
 	}
 
+	@ParameterizedTest
+	@CsvSource({ "https://subdomain.accounts400.ondemand.com#anyFragment_keys",
+			"https://subdomain.accounts400.ondemand.com?a=b",
+			"\0://myauth.com",
+			"https://otherDomain.org?accounts400.ondemand.com",
+			"subdomain.accounts400.ondemand.com",})
+	void validationFails_xsuaaToken(String issuer) {
+		cut = new JwtIssuerValidator(Arrays.asList(domains));
+		configureMock(issuer, null);
+		when(token.getService()).thenReturn(Service.XSUAA);
+
+		ValidationResult validationResult = cut.validate(token);
+		assertThat(validationResult.isErroneous(), is(true));
+		assertThat(validationResult.getErrorDescription(), startsWith("Issuer is not trusted because issuer "));
+	}
+
+	@ParameterizedTest
+	@CsvSource({ "https://otherDomain.accounts400.ondemand.com",
+			"https://paas.accounts400.ondemand.com", })
+	void validationSucceeds_XsuaaToken(String issuer) {
+		cut = new JwtIssuerValidator(Arrays.asList(domains));
+		configureMock(issuer, null);
+		when(token.getService()).thenReturn(Service.XSUAA);
+		ValidationResult validationResult = cut.validate(token);
+		assertThat(validationResult.isValid(), is(true));
+	}
+
 	@ParameterizedTest
 	@CsvSource({ "https://otherDomain.accounts400.ondemand.com,",
-			"https://paas.accounts400.ondemand.com,",
-			"https://nestle.com,https://paas.accounts400.ondemand.com," })
-	void validationSucceeds(String issuer, String iasIssuer) {
+			"subdomain.accounts400.ondemand.com,",
```

Comment:
I haven't understood, why you have changed the parameterized test case instead of adding another one?

Reply:
I've reverted the overwrite and added your new parameterized test cases at the end...